File: leetcode/3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros/solution.py
class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        #sums1 = 4, sums2 = 5 nums with 0 bigger than the other by 1*# zeros or more
        #sums1, sums2, #zeros in nums1, nums2
        
        #1. check the number of zeros and get sums
        #2. if one array doesn't have 0s and < the other array after replacing all 0's with 1's, reteurn -1
    #3. replace 0's with 1's and return the bigger sums

        count1 = nums1.count(0)
        count2 = nums2.count(0)

        sums1 = sum(nums1)
        sums2 = sum(nums2)
        
        min1 = sums1 + count1
        min2 = sums2 + count2

        if count1 == 0 and min2 > sums1:
            return -1
        if count2 == 0 and min1 > sums2:
            return -1

        return max(min1, min2)

File: leetcode/3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros/solution.cpp
class Solution {
public:
    long long minSum(vector<int>& nums1, vector<int>& nums2) {
        long count1 = 0;
        long count2 = 0;
        long sums1 = 0;
        long sums2 = 0;
        
        for(int i: nums1) {
            sums1 += i;
            if (i == 0) {
                count1++;
            }
        }

        for(int i: nums2) {
            sums2 += i;
            if (i == 0) {
                count2++;
            }
        }

        long min1 = sums1 + count1;
        long min2 = sums2 + count2;

        if (count1 == 0 && min2 > min1) {
            return -1;
        }
        if (count2 == 0 && min1 > min2) {
            return -1;
        }

        return max(min1, min2);
    }
};
